docs/buildah-bud.md

```diff
@@ -339,16 +339,12 @@ not required for Buildah as it supports only Linux.
 
 **--pull**
 
-When the flag is enabled, attempt to pull the latest image from the registries
-listed in registries.conf if a local image does not exist or the image is newer
-than the one in storage. Raise an error if the image is not in any listed
-registry and is not present locally.
+Pull image before building ("always"|"missing"|"never") (default "missing").
```

Comment:
This is a change.  The --pull used to mean to look at the registry and pull if newer.  Now that is not an option, as I understand it.
Should their be another option "newer", which should be default and pull a newer image if it exists at the registy otherwise use local.

Reply:
If current boolean --pull also for pulling the newer image, I need to figure out to prevent buildah from pulling new image when `--pull missing` the image exists locally but has a newer version in the registry.

Reply:
Does buildah  `--pull` can pull the newer image and replace the old one in the local? I tried the master buildah didn't see the update and didn't find the code implemented this functionality.
@TomSweeneyRedHat Do you know which PR I can refer to for finding newer version image from registry?

Reply:
`buildah bud --pull` is supposed to pull the image if it doesn't exist locally or if the version on the registry is newer there than the version stored locally.  I've not tested it recently to verify that it does do that.

Where it happens is fun to find.

https://github.com/containers/buildah/blob/master/new.go#L100 is where resolveImage() is called and the check happens here:  https://github.com/containers/buildah/blob/master/new.go#L140

Or at least that's the theory.

Reply:
thanks, @TomSweeneyRedHat but not code there, that code is for `--pull-always.`
if bud --pull, buildah will set options.PullPolicy = PullIfMissing, but https://github.com/containers/buildah/blob/master/new.go#L140 is inside the `if options.PullPolicy == PullAlways`